app/controllers/responses_controller.rb

```diff
@@ -218,4 +217,12 @@ def render_xml_submission_failure(exception, code)
       Rails.logger.info("XML submission failed: '#{exception.to_s}'")
       render(:nothing => true, :status => code)
     end
+
+    def response_params
+      if params[:response]
+        params.require(:response).permit(:form_id, :user_id, :incomplete, :reviewed).tap do |whitelisted|
+          whitelisted[:answers_attributes] = params[:response][:answers_attributes]
+        end
```

Comment:
So here you're basically using `permit!` without actually using it. Not sure if this is a good idea as it defeats the purpose of SP.

How come we can't use regular SP? The keys under answers_attributes are known, are they not?